value-processor/src/org/immutables/value/processor/Immutables.generator

```diff
@@ -697,21 +697,23 @@ public [type.typeAbstract.relative] [v.names.with]([v.atNullability][v.type] val
 [else if type.useSingletonOnly][-- deliberately left empty --]
 [else if type.useInterned]
 
+  private static final class InternerHolder {
```

Comment:
We're effectively applying the [Initialization-on-demand holder idiom](https://en.wikipedia.org/wiki/Initialization-on-demand_holder_idiom).